### PR TITLE
[All hosts] Debugging: Fix steps for debugging without yo office

### LINKED ISSUE #4376: sample launch.config in appendix A appears to use wrong "type"
The example `launch.config` here specifies `type: "pwa-msedge"`
this displays an error in the vscode editor, and suggests using type msedge instead.

which should it be?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: d5e92a7f-e42d-68ba-cc21-15df8289f5a1
* Version Independent ID: 096a172b-382e-764d-6d7d-88811bb5b96f
* Content: [Debug add-ins on Windows using Visual Studio Code and Microsoft Edge WebView2 (Chromium-based) - Office Add-ins](https://learn.microsoft.com/en-us/office/dev/add-ins/testing/debug-desktop-using-edge-chromium)
* Content Source: [docs/testing/debug-desktop-using-edge-chromium.md](https://github.com/OfficeDev/office-js-docs-pr/blob/main/docs/testing/debug-desktop-using-edge-chromium.md)
* Service: **microsoft-365**
* Sub-service: **add-ins**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

### PR DESCRIPTION
**Pull request by davidchesnut**
This rewrites the appendices into a procedure on how to debug any plain project (non yo office) using VS Code.

Fixes #4376

### CODE REVIEW

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 1]
```diff
@@ -33,18 +34,18 @@ These instructions assume you have experience using the command line, understand
...
 
 1. From the **RUN AND DEBUG** options, choose the Edge Chromium option for your host application, such as **Outlook Desktop (Edge Chromium)**. Select **F5** or choose **Run** > **Start Debugging** from the menu to begin debugging. This action automatically launches a local server in a Node window to host your add-in and then automatically opens the host application, such as Excel or Word. This may take several seconds.
 
    > [!TIP]
    > If you aren't using a project created with Yo Office, you may be prompted to adjust a registry key. While in the root folder of your project, run the following in the command line.
    >
-   > ``` command&nbsp;line
+   > ```command line
    > npx office-addin-debugging start <your manifest path>
    > ```
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:51]: There's still a reference to Appendix B in the next section. Seems like maybe that whole important note could be removed now?

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 2]
```diff
@@ -33,18 +34,18 @@ These instructions assume you have experience using the command line, understand
 
    - If you want to create a project to experiment with debugging in Visual Studio Code, use the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md). Use any one of our quick start guides, such as the [Outlook add-in quick start](../quickstarts/outlook-quickstart.md), in order to do this.
    - If you want to debug an existing project that was created with Yo Office, skip to the next step.
-   - If you want to debug an existing project that was not created with Yo Office, complete the procedure in the [Appendix A](#appendix-a) and then return to the next step of this procedure.
+   - If you want to debug an existing project that was not created with Yo Office, complete the procedure in the [Debug a project not created with yo office](#debug-a-project-not-created-with-yo-office) and then return to the next step of this procedure.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:37]: The text before the numbered steps mentions a yo office project as a prerequisite. Seems confusing to have the instructions about what to do if that isn't the case embedded later.

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 3]
```diff
@@ -33,18 +34,18 @@ These instructions assume you have experience using the command line, understand
...
 
-1. Choose  **View** > **Run** or enter **Ctrl+Shift+D** to switch to debug view.
+1. Choose **View** > **Run** or enter **Ctrl+Shift+D** to switch to debug view.
 
 1. From the **RUN AND DEBUG** options, choose the Edge Chromium option for your host application, such as **Outlook Desktop (Edge Chromium)**. Select **F5** or choose **Run** > **Start Debugging** from the menu to begin debugging. This action automatically launches a local server in a Node window to host your add-in and then automatically opens the host application, such as Excel or Word. This may take several seconds.
 
    > [!TIP]
    > If you aren't using a project created with Yo Office, you may be prompted to adjust a registry key. While in the root folder of your project, run the following in the command line.
    >
-   > ``` command&nbsp;line
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:47]: Do we no longer have to write command line this way? When did that change?

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
I haven't tested this, but I had a few comments from reading through the article.

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 4]
```diff
@@ -1,9 +1,10 @@
 ---
 title: Debug add-ins on Windows using Visual Studio Code and Microsoft Edge WebView2 (Chromium-based)
 description: Learn how to debug Office Add-ins that use Microsoft Edge WebView2 (Chromium-based) in VS Code.
-ms.date: 08/18/2023
+ms.date: 01/26/2024
 ms.localizationpriority: high
 ---
+
 # Debug add-ins on Windows using Visual Studio Code and Microsoft Edge WebView2 (Chromium-based)
```
Comment (lindalu-MSFT) [inline docs/testing/debug-desktop-using-edge-chromium.md:8]:
```suggestion
# Debug Office Add-ins on Windows using Visual Studio Code and Microsoft Edge WebView2 (Chromium-based)
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 5]
```diff
@@ -1,9 +1,10 @@
 ---
 title: Debug add-ins on Windows using Visual Studio Code and Microsoft Edge WebView2 (Chromium-based)
```
Comment (lindalu-MSFT) [inline docs/testing/debug-desktop-using-edge-chromium.md:2]:
```suggestion
title: Debug Office Add-ins on Windows using Visual Studio Code and Microsoft Edge WebView2 (Chromium-based)
```
For SEO

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 6]
```diff
@@ -33,18 +34,18 @@ These instructions assume you have experience using the command line, understand
 
    - If you want to create a project to experiment with debugging in Visual Studio Code, use the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md). Use any one of our quick start guides, such as the [Outlook add-in quick start](../quickstarts/outlook-quickstart.md), in order to do this.
```
Comment (lindalu-MSFT) [inline docs/testing/debug-desktop-using-edge-chromium.md:35]:
```suggestion
   - If you want to create a project to experiment with debugging in Visual Studio Code, use the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md). To do this, use any of our quick start guides, such as the [Outlook add-in quick start](../quickstarts/outlook-quickstart.md).
```

**Review by lindalu-MSFT — APPROVED** [review]
A couple SEO suggestions and one grammar nit on a section you didn't update! Otherwise, looks fine. I did not test the steps, however.

File: docs/testing/debug-desktop-using-edge-chromium.md (hunk 2 above)
Comment (davidchesnut) [reply docs/testing/debug-desktop-using-edge-chromium.md:37]: I removed this step and modified the titles to indicate there are 2 sections (yo office, and non-yo office)

File: docs/testing/debug-desktop-using-edge-chromium.md (hunk 3 above)
Comment (davidchesnut) [reply docs/testing/debug-desktop-using-edge-chromium.md:47]: Hmmm, not sure.

File: docs/testing/debug-desktop-using-edge-chromium.md (hunk 1 above)
Comment (davidchesnut) [reply docs/testing/debug-desktop-using-edge-chromium.md:51]: Restored Appendix B (didn't realize it was for this error scenario).

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 7]
```diff
@@ -25,26 +26,25 @@ This debugging mode is dynamic, allowing you to set breakpoints while code is ru
 - Windows 10, 11
 - A combination of platform and Office application that supports Microsoft Edge with WebView2 (Chromium-based) as explained in [Browsers and webview controls used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md). If your version of Office from a Microsoft 365 subscription is earlier than Version 2101, you will need to install WebView2. Use the instructions for installing it at [Microsoft Edge WebView2 / Embed web content ... with Microsoft Edge WebView2](https://developer.microsoft.com/microsoft-edge/webview2/).
```
Comment (lindalu-MSFT) [inline docs/testing/debug-desktop-using-edge-chromium.md:27]:
```suggestion
- A combination of platform and Office application that supports Microsoft Edge with WebView2 (Chromium-based) as explained in [Browsers and webview controls used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md). If your version of Office from a Microsoft 365 subscription is earlier than Version 2101, you'll need to install WebView2. For instructions to install WebView2, see [Microsoft Edge WebView2 / Embed web content ... with Microsoft Edge WebView2](https://developer.microsoft.com/microsoft-edge/webview2/).
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 8]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
 >
 > If debugging stops working; for example, if breakpoints are being ignored; stop debugging. Then, if necessary, close all host application windows and the Node window. Finally, close Visual Studio Code and reopen it.
```
Comment (lindalu-MSFT) [inline docs/testing/debug-desktop-using-edge-chromium.md:79]:
```suggestion
> If debugging stops working---for example, if breakpoints are being ignored---stop debugging. Then, if necessary, close all host application windows and the Node window. Finally, close Visual Studio Code and reopen it.
```
Use em dash instead of semicolon here! The triple hyphen is supposed to render as em dash in GitHub flavored markdown. Alternatively, you could insert HTML here. Both **&mdash;**  and  **&#151;** render an em dash.

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 9]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
-   
      // Other properties may be here.
    }
    ```
 
-1. Add the following object to the `configurations` array.
+   > [!NOTE]
+   > If you already have a launch.json file, just add the single configuration to the `configurations` section.
+
+1. Replace the placeholder `$HOST$` with the name of the Office application that the add-in runs in; for example, `Outlook` or `Word`.
```
Comment (lindalu-MSFT) [inline docs/testing/debug-desktop-using-edge-chromium.md:127]:
```suggestion
1. Replace the placeholder `$HOST$` with the name of the Office application that the add-in runs in. For example, `Outlook` or `Word`.
```
Semicolon = conjunction. For example, and, but, or or.

**Review by lindalu-MSFT — COMMENTED** [review]
I corrected one contraction and committed it already thinking there wouldn't be any edits here, but then I saw the semicolon misused a couple of times and decided to start a review! Other than these two edits, everything else looks good.

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 10]
```diff
@@ -23,28 +24,27 @@ This debugging mode is dynamic, allowing you to set breakpoints while code is ru
...
 
-## Use the Visual Studio Code debugger
+## Debug a project created with Yo Office
 
 These instructions assume you have experience using the command line, understand basic JavaScript, and have created an Office Add-in project before using the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md). If you haven't done this before, consider visiting one of our tutorials, such as the [Excel Office Add-in tutorial](../tutorials/excel-tutorial.md).
 
 1. The first step depends on the project and how it was created.
 
-   - If you want to create a project to experiment with debugging in Visual Studio Code, use the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md). Use any one of our quick start guides, such as the [Outlook add-in quick start](../quickstarts/outlook-quickstart.md), in order to do this.
+   - If you want to create a project to experiment with debugging in Visual Studio Code, use the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md). To do this, use any of our quick start guides, such as the [Outlook add-in quick start](../quickstarts/outlook-quickstart.md).
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:35]: A suggestion to consider that avoids two sentences in a row with the pattern "X, use Y".
```suggestion
   - If you want to create a project to experiment with debugging in Visual Studio Code, use the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md). Follow any of the yo office quick start guides, such as the [Outlook add-in quick start](../quickstarts/outlook-quickstart.md).
```

File: docs/testing/debug-desktop-using-edge-chromium.md (hunk 3 above)
Comment (AlexJerabek) [reply docs/testing/debug-desktop-using-edge-chromium.md:47]: If it renders fine, great! Maybe double check after this gets merged. If all is well, we should change that everywhere.

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 11]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
 
-### Appendix A
+## Debug a project not created with Yo Office
 
 If your project wasn't created with Yo Office, you need to create a debug configuration for Visual Studio Code.
 
+### Configure package.json file
+
+1. Ensure you have a `package.json` file. If you don't already have a package.json file, run `npm init` in the root folder of your project and answer the prompts.
+1. Run `npm install office-addin-debugging`. This package will sideload your add-in for debugging.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:88]: Nit. Personal preference to avoid future tense in step-by-step instructions.
```suggestion
1. Run `npm install office-addin-debugging`. This package sideloads your add-in for debugging.
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 12]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
+1. Open the package.json file. In the `scripts` section add the following script.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:89]: Consistency with the earlier code ticks around the file name.
```suggestion
1. Open the `package.json` file. In the `scripts` section, add the following script.
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 13]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
+
+   ```json
+   "start:desktop": "office-addin-debugging start $MANIFEST_FILE$ desktop",
+   "dev-server": "$SERVER_START$"
+   ```
+
+1. Replace `$MANIFEST_FILE$` with the correct file name and folder location of your manifest.xml file.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:96]: Since the previous parts of the instruction allude to this file name being flexible, consider not using the standard file name here.
```suggestion
1. Replace `$MANIFEST_FILE$` with the correct file name and folder location of your manifest.
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 14]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
+1. Replace `$SERVER_START$` with the command to start your web server. Later in these steps, the office-addin-debugging package will specifically look for the `dev-server` script to launch your web server.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:97]: Consistent code ticks about package names.
```suggestion
1. Replace `$SERVER_START$` with the command to start your web server. Later in these steps, the `office-addin-debugging` package will specifically look for the `dev-server` script to launch your web server.
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 15]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
+       }
      ]
-   
      // Other properties may be here.
    }
    ```
 
-1. Add the following object to the `configurations` array.
+   > [!NOTE]
+   > If you already have a launch.json file, just add the single configuration to the `configurations` section.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:125]:
```suggestion
   > If you already have a `launch.json` file, just add the single configuration to the `configurations` section.
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 16]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
+
+1. Replace the placeholder `$HOST$` with the name of the Office application that the add-in runs in; for example, `Outlook` or `Word`.
+1. Save and close the file.
+
+### Configure tasks.json
+
+1. Create a file named `tasks.json` in the `\.vscode` folder of the project.
+1. Copy the following JSON into the file. It contains a task that will start debugging for your add-in.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:133]:
```suggestion
1. Copy the following JSON into the file. It contains a task that starts debugging for your add-in.
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 17]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
+         "problemMatcher": []
+       }
+     ]
+   }
    ```
 
-1. Replace the placeholder `$HOST$` with the name of the Office application that the add-in runs in; for example, `Outlook` or `Word`.
-1. Save and close the file.
+   > [!NOTE]
+   > If you already have a tasks.json file, just add the single task to the `tasks` section.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:155]:
```suggestion
   > If you already have a `tasks.json` file, just add the single task to the `tasks` section.
```

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 18]
```diff
@@ -78,45 +78,85 @@ These instructions assume you have experience using the command line, understand
...
+
+1. Replace both instances of the placeholder `$HOST$` with the name of the Office application that the add-in runs in; for example, `Outlook` or `Word`.
```
Comment (AlexJerabek) [inline docs/testing/debug-desktop-using-edge-chromium.md:157]: That semicolon pattern Linda called out is here too. Make sure whatever changes you make are propagated.

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
Content looks good. Just some style things.

File: docs/testing/debug-desktop-using-edge-chromium.md [hunk 19]
```diff
@@ -76,47 +76,87 @@ These instructions assume you have experience using the command line, understand
...
+     ]
+   }
    ```
 
-1. Replace the placeholder `$HOST$` with the name of the Office application that the add-in runs in; for example, `Outlook` or `Word`.
-1. Save and close the file.
+   > [!NOTE]
+   > If you already have a `tasks.json` file, just add the single task to the `tasks` section.
+
+1. Replace both instances of the placeholder `$HOST$` with the name of the Office application that the add-in runs in; for example, `Outlook` or `Word`.
```
Comment (davidchesnut) [inline docs/testing/debug-desktop-using-edge-chromium.md:157]:
```suggestion
1. Replace both instances of the placeholder `$HOST$` with the name of the Office application that the add-in runs in. For example, `Outlook` or `Word`.
```

**Review by AlexJerabek — APPROVED** [review]
